src/react-app.js

```diff
@@ -77,23 +76,27 @@ function renderReactApp({
   render,
 }) {
   const exerciseInfo = []
-  const exerciseTypes = ['final', 'exercise', 'extraCredit', 'instruction']
+  const exerciseTypes = ['final', 'exercise', 'instruction']
+  filesInfo.sort((a, b) => {
+    // change order so that shorter file names (01) are before longer (01.extra-01)
+    if (a.filename.includes(b.filename)) return 1
+    if (b.filename.includes(a.filename)) return -1
+    // otherwise preserve existing order from glob
+    return 0
+  })
```

Comment:
Could we make this sorting happen when we do the codegen so they're in the right order by the time it gets here?

Reply:
I will do this a bit later

Reply:
sort is now happening during codegen and the snapshot is updated